frigate/comms/dispatcher.py

```diff
@@ -0,0 +1,193 @@
+"""Handle communication between frigate and other applications."""
+
+import logging
+
+from typing import Any, Callable
+
+from abc import ABC, abstractmethod
+
+from frigate.config import FrigateConfig
+from frigate.types import CameraMetricsTypes
+from frigate.util import restart_frigate
+
+
+logger = logging.getLogger(__name__)
+
+
+class Communicator(ABC):
+    """pub/sub model via specific protocol."""
+
+    @abstractmethod
+    def publish(self, topic: str, payload: Any, retain: bool = False) -> None:
+        """Send data via specific protocol."""
+        pass
+
+    @abstractmethod
+    def subscribe(self, receiver: Callable) -> None:
+        """Pass receiver so communicators can pass commands."""
+        pass
+
+
+class Dispatcher:
+    """Handle communication between frigate and communicators."""
+
+    def __init__(
+        self,
+        config: FrigateConfig,
+        camera_metrics: dict[str, CameraMetricsTypes],
+        communicators: list[Communicator],
+    ) -> None:
+        self.config = config
+        self.camera_metrics = camera_metrics
+        self.comms = communicators
+
+        for comm in self.comms:
+            comm.subscribe(self._receive)
+
+        self._camera_settings_handlers: dict[str, Callable] = {
+            "detect": self._on_detect_command,
+            "improve_contrast": self._on_motion_improve_contrast_command,
+            "motion": self._on_motion_command,
+            "motion_contour_area": self._on_motion_contour_area_command,
+            "motion_threshold": self._on_motion_threshold_command,
+            "recording": self._on_recordings_command,
+            "snapshots": self._on_snapshots_command,
+        }
+
+    def _receive(self, topic: str, payload: str) -> None:
+        """Handle receiving of payload from communicators."""
+        if topic.endswith("set"):
+            camera_name = topic.split("/")[-3]
```

Comment:
Let's wrap these in a try/except so errant mqtt messages can't crash this thread.